x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/persistence/JobConfigProvider.java

```diff
@@ -376,21 +411,131 @@ private QueryBuilder buildQuery(String expression) {
         return boolQueryBuilder;
     }
 
-    private void parseJobLenientlyFromSource(BytesReference source, ActionListener<Job.Builder> jobListener)  {
-        try (InputStream stream = source.streamInput();
-             XContentParser parser = XContentFactory.xContent(XContentType.JSON)
-                     .createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, stream)) {
-            jobListener.onResponse(Job.LENIENT_PARSER.apply(parser, null));
-        } catch (Exception e) {
-            jobListener.onFailure(e);
+    /**
+     * Does the {@code tokens} array resolves to a wildcard all expression.
+     * True if {@code tokens} is empty or if it contains a single element
```

Comment:
If the tokens contain a full wildcard and something else then that's still a full wildcard overall.  E.g. `foo,*` still means get everything, but with the way this method is used it makes sense not to report it.  I think the comment should be changed to make clear that this function doesn't cover this case.